File: SelectionSort/CPP/SelectionSort.cpp
//
// Created by Shahriar Nasim Nafi on 28/6/22.
//

void selectionSort(int *array, int size) {
    int minIdx = 0;
    for (int i = 0; i < size - 1; i++) {
        minIdx = i;
        for (int j = i + 1; j < size; j++) {
            if (array[j] < array[minIdx]) {
                minIdx = j;
            }
        }

        if (minIdx != i) {
            int temp = array[minIdx];
            array[minIdx] = array[i];
            array[i] = temp;
        }
    }
}

File: SelectionSort/Swift/SelectionSort.swift
//
// Created by Shahriar Nasim Nafi on 28/6/22.
//

extension Array where Element: Comparable {
    func selectionSort() -> [Element] {
        var sorted = self
        var minIdx: Int
        for i in 0..<sorted.count - 1 {
            minIdx = i
            for j in (i+1)..<sorted.count {
                if sorted[j] < sorted[minIdx] {
                    minIdx = j
                }
            }
            if minIdx != i {
                (sorted[minIdx], sorted[i]) = (sorted[i], sorted[minIdx])
            }
        }
        return sorted
    }
}
